### PR TITLE
Add Elastic Agent configuration examples

### PR DESCRIPTION
**Pull request by david-kow**
This PR adds:
- configuration examples for enabling System and Kubernetes integrations
- README describing the examples
- example specific E2E tests
- utilities needed to run those tests

We might add more examples (eg. multiple outputs) depending on features added and readiness on Elastic Agent side.

Exposing configuration examples via docs will be a part of https://github.com/elastic/cloud-on-k8s/issues/4011.

Resolves https://github.com/elastic/cloud-on-k8s/issues/4017.

### CODE REVIEW

**Comment by david-kow** [comment]
run full pr build

File: config/recipes/elastic-agent/kubernetes-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      - id: kubernetes/metrics-kubernetes.volume
+        data_stream:
+          dataset: kubernetes.volume
+          type: metrics
+        metricsets:
+        - volume
+        add_metadata: true
+        bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+        hosts:
+        - 'https://${NODE_NAME}:10250'
```
Comment (pebrc) [inline config/recipes/elastic-agent/kubernetes-integration.yaml:124]: Should we add a disclaimer that this is not going to work on OCP ?

**Review by pebrc — APPROVED** [review]
LGTM! Disclaimer: I did not run the examples as e2e tests myself only manually.

File: config/recipes/elastic-agent/kubernetes-integration.yaml (hunk 1 above)
Comment (david-kow) [reply config/recipes/elastic-agent/kubernetes-integration.yaml:124]: I'd think we say that already: 

> These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster.

I'd consider OCP to be a secured as it required SCCs to run anything. Should we be more explicit than that?

**Comment by david-kow** [comment]
I'm merging this now. @pebrc, please let me know if I should improve https://github.com/elastic/cloud-on-k8s/pull/4030#discussion_r544053146 - if yes, I'll do that in a separate PR.

**Pull request merged**